### PR TITLE
Dropdown options for social media and instant messaging updated, placeholders changed to username

### PR DESCRIPTION
**Pull request by xh3n1**
Issues #130,  #252 #216

### CODE REVIEW

File: js/services/vCardProperties.js [hunk 1]
```diff
@@ -131,7 +131,10 @@ angular.module('contactsApp')
 			},
 			options: [
 				{id: 'FACEBOOK', name: 'Facebook'},
+				{id: 'INSTAGRAM', name: 'Instagram'},
+				{id: 'LINKEDIN', name: 'Linkedin'},
```
Comment (MariusBluem) [inline js/services/vCardProperties.js]: It’s spelled Linked*I*n 😉

File: js/services/vCardProperties.js [hunk 2]
```diff
@@ -91,15 +91,16 @@ angular.module('contactsApp')
...
 			defaultValue: {
 				value:[''],
-				meta:{type:['HOME']}
+				meta:{type:['SKYPE']}
 			},
 			options: [
-				{id: 'HOME', name: t('contacts', 'Home')},
-				{id: 'WORK', name: t('contacts', 'Work')},
-				{id: 'OTHER', name: t('contacts', 'Other')}
+				{id: 'IRC', name: t('contacts', 'Irc')},
```
Comment (MorrisJobke) [inline js/services/vCardProperties.js]: I would say, that those don't need to be translated. Also I would write "IRC", because it is an abbreviation.

File: js/services/vCardProperties.js [hunk 3]
```diff
@@ -91,15 +91,16 @@ angular.module('contactsApp')
...
+				{id: 'SKYPE', name: t('contacts', 'Skype')},
+		    {id: 'TELEGRAM', name: t('contacts', 'Telegram')}
```
Comment (MorrisJobke) [inline js/services/vCardProperties.js]: Please use the same indentation as for the other entries. Looks like a space vs tabs mixup. ;)

File: package-lock.json [hunk 4]
```diff
@@ -0,0 +1,4861 @@
+{
```
Comment (nickvergessen) [inline package-lock.json]: I don't think this is intended?

Comment (nickvergessen) [inline package-lock.json]: Or why is this file here all of a sudden?

File: js/services/vCardProperties.js [hunk 5]
```diff
@@ -91,15 +91,18 @@ angular.module('contactsApp')
...
 			defaultValue: {
 				value:[''],
-				meta:{type:['HOME']}
+				meta:{type:['SKYPE']}
 			},
 			options: [
-				{id: 'HOME', name: t('contacts', 'Home')},
-				{id: 'WORK', name: t('contacts', 'Work')},
-				{id: 'OTHER', name: t('contacts', 'Other')}
+				{id: 'IRC', name: t('contacts', 'IRC')},
```
Comment (MorrisJobke) [inline js/services/vCardProperties.js]: Could you change `t('contacts', 'IRC')` to `'IRC'` only - the translation is not really needed for those strings.

File: js/services/vCardProperties.js [hunk 6]
```diff
@@ -91,16 +91,16 @@ angular.module('contactsApp')
...
 			},
 			options: [
-				{id: 'HOME', name: t('contacts', 'Home')},
-				{id: 'WORK', name: t('contacts', 'Work')},
-				{id: 'OTHER', name: t('contacts', 'Other')}
-			]
+				{id: 'IRC', name:  'IRC'},
+				{id: 'SKYPE', name:'Skype'},
+				{id: 'TELEGRAM', name:'Telegram'}
+    	]
```
Comment (MorrisJobke) [inline js/services/vCardProperties.js]: "103:2  error  Mixed spaces and tabs  no-mixed-spaces-and-tabs" <- this is the travis error

**Comment by jancborchardt** [comment]
Awesome! Can you review @nextcloud/contacts @nextcloud/javascript? :)

Also cc @jpvdgiessen @qwertygc as it fixes issues you reported. Please review :)

File: js/services/vCardProperties.js [hunk 7]
```diff
@@ -124,14 +127,17 @@ angular.module('contactsApp')
...
 			defaultValue: {
 				value:[''],
 				meta:{type:['facebook']}
 			},
 			options: [
 				{id: 'FACEBOOK', name: 'Facebook'},
+				{id: 'INSTAGRAM', name: 'Instagram'},
+				{id: 'LINKEDIN', name: 'LinkedIn'},
 				{id: 'TWITTER', name: 'Twitter'}
+
```
Comment (jpvdgiessen) [inline js/services/vCardProperties.js]: Maybe also adding Google+ and Pinterest?

Comment (MariusBluem) [inline js/services/vCardProperties.js]: Should be the same options of socialsharing-app (https://github.com/nextcloud/socialsharing) ... So Google+ is a good idea 👍

Comment (xh3n1) [inline js/services/vCardProperties.js]: What about Reddit, Snapchat ? @MariusBluem

Comment (blizzz) [inline js/services/vCardProperties.js]: GnuSocial,  Diaspora :)

**Comment by blizzz** [comment]
This "username"  thing is visible? I hope it is filled with the user's display name?

**Comment by jancborchardt** [comment]
@xh3n1 maybe check what Google Contacts and Apple Contacts offer in their selection - we should have all those in our list for sure. :) Anything else we can still add in the future.

File: js/services/vCardProperties.js (hunk 7 above)
Comment (jancborchardt) [inline js/services/vCardProperties.js]: As said in the other comment let’s first start with the basics, not all and every single service we can think of. :) We can always extend later, but let’s not make this pull request too big.

**Comment by jancborchardt** [comment]
@jonatoni @xh3n1 here are the entries iCloud shows.

For »Social network«:
screenshot from 2017-07-10 11-38-07
Only one thing: Let’s add Instagram since that’s very popular.
All entries have »Username« as placeholder, except Twitter which prepends an @ as you can see.

This is for »Instant messaging«:
screenshot from 2017-07-10 11-38-42
All entries have »Username« as placeholder.

If those are all in the lists, we can merge the pull request. :)

**Comment by blizzz** [comment]
ICQ is still a thing?

**Comment by qwertygc** [comment]
We can add mastodon ?

**Comment by blizzz** [comment]
@qwertygc cf. https://github.com/nextcloud/contacts/pull/254#discussion_r126375824

> As said in the other comment let’s first start with the basics, not all and every single service we can think of. :) We can always extend later, but let’s not make this pull request too big.

**Comment by jancborchardt** [comment]
So can we go ahead and merge it like this? Further services can be added in a future pull request. :)

@irgendwie @Henni @skjnldsv does one of you mind giving it a quick check?

**Comment by Henni** [comment]
This should still be addressed: https://github.com/nextcloud/contacts/pull/254#discussion_r126141921
Otherwise code looks good (currently not able to actually test it).

**Comment by jancborchardt** [comment]
Whitespace comment of @MorrisJobke @Henni was addressed, merging. :)

Congrats on your first pull request @xh3n1 @jonatoni! :)

**Comment by jancborchardt** [comment]
@xh3n1 @jonatoni I added a comment with the remaining services to add at https://github.com/nextcloud/contacts/issues/130#issuecomment-315046090 :)
